src/eval.rs

```diff
@@ -26,15 +28,73 @@ pub enum RuntimeError {
 }
 
 // abstraction for values making up expressions
-#[derive(Debug, Clone, PartialEq)]
+
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Value {
     Bool(bool),
     Num(i64, Range<i64>),
     Sym(Symbol),
 }
 
+impl PartialOrd for Value {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        use Value::*;
+        Some(match (self, other) {
+            // Bool comparison: false < true (default Rust behavior)
+            (Bool(a), Bool(b)) => a.cmp(b),
+
+            // Bool < Num < Sym
+            (Bool(_), Num(_, _)) | (Bool(_), Sym(_)) => Ordering::Less,
+            (Num(_, _), Bool(_)) | (Sym(_), Bool(_)) => Ordering::Greater,
+
+            // Num comparison: by value, then range.start, then range.end
+            (Num(a_val, a_rng), Num(b_val, b_rng)) => a_val
+                .cmp(b_val)
+                .then(a_rng.start.cmp(&b_rng.start))
+                .then(a_rng.end.cmp(&b_rng.end)),
+
+            // Num < Sym
+            (Num(_, _), Sym(_)) => Ordering::Less,
+            (Sym(_), Num(_, _)) => Ordering::Greater,
+
+            // Sym comparison
+            (Sym(a), Sym(b)) => a.cmp(b),
+        })
+    }
+}
+
+impl fmt::Display for Value {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Value::Bool(b) => write!(f, "{}", b),
+            Value::Num(n, range) => write!(f, "{} in [{}..{})", n, range.start, range.end),
+            Value::Sym(sym) => write!(f, "{}", sym),
+        }
+    }
+}
+
+impl Ord for Value {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.partial_cmp(other).unwrap()
+    }
+}
+
+pub fn eval_action(program: &Program, env: &mut Env, sym: &Symbol) -> Result<(), RuntimeError> {
+    // Insert each symbol into the environment
```

Comment:
the comments here are unnecessary and somewhat misleading, it is not inserting "each" symbol.